distributed/utils.py

```diff
@@ -470,62 +518,53 @@ def start(self):
         with self._lock:
             self._start_unlocked()
 
-    def _start_unlocked(self):
+    def _start_unlocked(self) -> None:
         assert not self._started
 
-        count, real_runner = self._all_loops[self._loop]
-        if self._asynchronous or real_runner is not None or count > 0:
+        if self._loop is not None:
+            try:
+                count, real_runner = self._all_loops[self._loop]
+            except KeyError:
+                assert self._loop.asyncio_loop.is_running()  # type: ignore[attr-defined]
+                self._started = True
+                return
+
             self._all_loops[self._loop] = count + 1, real_runner
```

Comment:
If multiple `LoopRunner`s are spawned in separate threads, but all given the same loop, this mutation isn't thread-safe. (IDK if `LoopRunner` is supposed to be thread-safe in the first place.)

Reply:
This is run protected by a lock